File: 3BV2/PP/P1/Ejercicios/06-Polimorfismo/Python/Parte2/main.py
from estudiante import Estudiante
from docente import Docente
from paae import PAAE

# Crear instancias de Estudiante, Docente y PAAE
estudiante = Estudiante('1', 'Eliot', 'CURPELIOT1234', 'Calle 123')
docente = Docente('2', 'Jaime', 'CURPJAIME5678', 'Calle 456')
paae = PAAE('3', 'Itzel', 'CURPITZEL9012', 'Calle 789')

while True:
    print("\n\n=====================================================================")
    print('Por favor, selecciona una opción:')
    print('1. Estudiante')
    print('2. Docente')
    print('3. PAAE')
    print('4. Salir')

    opcion = int(input())  # Leer la opción seleccionada por el usuario

    if opcion == 1:
        persona_seleccionada = estudiante
    elif opcion == 2:
        persona_seleccionada = docente
    elif opcion == 3:
        persona_seleccionada = paae
    elif opcion == 4:
        print('Saliendo del programa...')
        break  # Salir del ciclo while y finalizar el programa
    else:
        print('Opción inválida')
        continue  # Volver al inicio del ciclo para solicitar nuevamente una opción

    # Mostrar la información de la persona seleccionada
    print("\n------------------")
    print('ID:', persona_seleccionada.get_id())
    print('Nombre:', persona_seleccionada.get_nombre())
    print('CURP:', persona_seleccionada.get_curp())
    print('Domicilio:', persona_seleccionada.get_domicilio())
    print()

File: 3BV2/PP/P1/Ejercicios/06-Polimorfismo/Java/Parte2/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        // Crear instancias de las personas (Estudiante, Docente, PAAE)
        Estudiante estudiante = new Estudiante("1", "Eliot", "CURPELIOT1234", "Calle 123");
        Docente docente = new Docente("2", "Jaime", "CURPJIMMY5678", "Calle 456");
        PAAE paae = new PAAE("3", "Itzel", "CURPITZEL9012", "Calle 789");

        // Crear un objeto Scanner para recibir la entrada del usuario
        Scanner scanner = new Scanner(System.in);
        int opcion = 0; // Variable para almacenar la opción seleccionada por el usuario

        // Bucle while para permitir al usuario elegir personas repetidamente hasta que ingrese la opción 4 para salir
        while (opcion != 4) {
            System.out.println("Por favor, selecciona una opción:");
            System.out.println("1. Estudiante");
            System.out.println("2. Docente");
            System.out.println("3. PAAE");
            System.out.println("4. Salir");

            opcion = scanner.nextInt(); // Leer la opción seleccionada por el usuario
            scanner.nextLine(); // Consumir el salto de línea después de leer la opción

            // Utilizar un switch para ejecutar el código correspondiente a la opción seleccionada
            switch (opcion) {
                case 1:
                    mostrarInformacion(estudiante); // Mostrar información del Estudiante
                    break;
                case 2:
                    mostrarInformacion(docente); // Mostrar información del Docente
                    break;
                case 3:
                    mostrarInformacion(paae); // Mostrar información del PAAE
                    break;
                case 4:
                    System.out.println("Saliendo del programa..."); // Mostrar mensaje de salida
                    break;
                default:
                    System.out.println("Opción inválida"); // Mostrar mensaje de opción inválida
                    break;
            }
        }

        scanner.close(); // Cerrar el objeto Scanner
    }

    // Método para mostrar la información de una persona en la consola
    public static void mostrarInformacion(Persona persona) {
        System.out.println("ID: " + persona.getID());
        System.out.println("Nombre: " + persona.getNombre());
        System.out.println("CURP: " + persona.getCURP());
        System.out.println("Domicilio: " + persona.getDomicilio());
        System.out.println();
    }
}
